app/javascript/components/layout/Header.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { Navbar, Nav, Form } from 'react-bootstrap';
+
+class Header extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    return (
+      <Navbar bg="primary" variant="dark" expand="lg" className="mb-3">
+        <Navbar.Brand href={this.props.report_mode ? '/' : this.props.root}>
+          Sara Alert<small className="nav-version ml-1">{this.props.version}</small>
+        </Navbar.Brand>
+        {this.props.current_user && (
+          <React.Fragment>
+            <Nav className="pt-1 mr-auto" activeKey={window.location.pathname}>
+              {this.props.current_user.can_see_enroller_dashboard_tab && (
+                <Nav.Link className="ml-3" href={`${window.BASE_PATH}/patients`}>
+                  <i className="fas fa-table"></i>&nbsp;&nbsp;Enroller Dashboard
+                </Nav.Link>
+              )}
+              {this.props.current_user.can_see_monitoring_dashboards_tab && (
+                <Nav.Link className="ml-3" href={`${window.BASE_PATH}/public_health`}>
+                  <i className="fas fa-table"></i>&nbsp;&nbsp;Monitoring Dashboards
+                </Nav.Link>
+              )}
+              {this.props.current_user.can_see_admin_panel_tab && (
+                <Nav.Link className="ml-3" href={`${window.BASE_PATH}/admin`}>
+                  <i className="fas fa-user-cog"></i>&nbsp;&nbsp;Admin Panel
+                </Nav.Link>
+              )}
+              {this.props.current_user.can_see_analytics_tab && (
+                <Nav.Link className="ml-3" href={`${window.BASE_PATH}/analytics`}>
+                  <i className="fas fa-chart-pie"></i>&nbsp;&nbsp;Analytics
+                </Nav.Link>
+              )}
+            </Nav>
+            <Form inline className="ml-auto">
+              {!this.props.report_mode && (
+                <React.Fragment>
+                  <Navbar.Text className="text-white px-3">
+                    <i className="fas fa-user"></i>&nbsp;&nbsp;{this.props.current_user.email}
```

Comment:
Sometimes in development, it's nice to browse directly to an assessment. There wouldn't be a current_user in that case, so it would be nice to make use of the `?` optional chaining operator here.

Reply:
We need to be 100% sure this will not cause JS errors, as that would make the entire component not render. One benefit of ERB is that at least it will always show up, regardless of JS errors.

Reply:
Not sure what you mean about this line in particular, it's the same as it was before in that regard: https://github.com/SaraAlert/SaraAlert/blob/master/app/views/layouts/_header.html.erb#L8

Happy to add the chaining operator (and will in case the email is nil somehow) - but just a note that there is a check on line 16 to make sure the current user does exist at this point already.

Agreed about being careful to make sure this component always renders.

Reply:
Added optional chaining operator to be safe.